src/edu/agh/wfiis/designpatterns/patternsUsageExample/patterns/DesignPattern.java

```diff
@@ -0,0 +1,77 @@
+package edu.agh.wfiis.designpatterns.patternsUsageExample.patterns;
+
+import edu.agh.wfiis.designpatterns.patternsUsageExample.enums.DesignPatternType;
+
+public class DesignPattern {
+
+    private String name;
+    private String patternCode;
+    private String description;
+    private String example;
+
+    private DesignPatternType type;
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getExample() {
+        return example;
+    }
+
+    public String getPatternCode() {
+        return patternCode;
+    }
+
+    public DesignPatternType getType() {
+        return type;
+    }
+
+    public boolean isTheSameTypeAs(DesignPattern patternToCompare) {
+        return type == patternToCompare.getType();
+    }
+
+    private DesignPattern(DesignPattern.Builder builder) {
+        this.name = builder.name;
+        this.description = builder.description;
+        this.example = builder.example;
+        this.patternCode = builder.patternCode;
+        this.type = builder.type;
+    }
+
+    public static class Builder {
+        private String name;
+        private String patternCode;
+        private String description;
+        private String example;
+
+        private DesignPatternType type;
+
+        public Builder(String name, String patternCode, DesignPatternType type){
```

Comment:
In simple types like DesignPattern it would be nice if Builder separated parameters of the same type with explicit indication what we set.

Like `DesignPattern.Builder().withName("name").withPatternName("patternName").build()` because now solves only problem with separation of mandatory and optional params. We still can accidentally swap `name` with `patternCode` during creating object.